sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -769,3 +769,41 @@ case object OneRowRelation extends LeafNode {
    */
   override lazy val statistics: Statistics = Statistics(sizeInBytes = 1)
 }
+
+/**
+ * An inline table that holds a number of foldable expressions, which can be materialized into
+ * rows. This is semantically the same as a Union of one row relations.
+ */
+case class InlineTable(rows: Seq[Seq[NamedExpression]]) extends LeafNode {
```

Comment:
Yeah, I had these checks. The thing is that none of the LogicalPlans have such logic, it has all been centralized in CheckAnalysis. So I added it there.

It might not be a bad plan to move this functionality into the separate plans on the longer run.